src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -313,94 +313,143 @@ protected static void DisplayManifestPreview(Manifests manifests)
         }
 
         /// <summary>
-        /// Validates the GitHubToken provided on the command-line, or if not present, the cached token if one exists.
-        /// Attempts a simple operation against the target repo, and if that fails, then:
-        ///     If token provided on command-line, errors out
-        ///     If not, and cached token was present, then deletes token cache, and starts OAuth flow
-        /// Otherwise, sets the instance variable to hold the validated token.
-        /// If no token is present on command-line or in cache, starts the OAuth flow to retrieve one.
+        /// If no GitHub token is provided, looks in the token cache and assigns it.
         /// </summary>
-        /// <param name="cacheToken">Boolean to override default behavior and force caching of token.</param>
-        /// <returns>True if the token is now present and valid, false otherwise.</returns>
-        protected async Task<bool> SetAndCheckGitHubToken(bool cacheToken = false)
+        protected async void CheckForGitHubToken()
         {
-            string cachedToken = null;
-            bool hasPatToken = !string.IsNullOrEmpty(this.GitHubToken);
-            string token = this.GitHubToken;
-
-            if (!hasPatToken)
+            if (string.IsNullOrEmpty(this.GitHubToken))
             {
                 Logger.Trace("No token parameter, reading cached token");
-                token = cachedToken = GitHubOAuth.ReadTokenCache();
+                this.GitHubToken = await this.GetTokenFromCache();
 
-                if (string.IsNullOrEmpty(token))
+                if (string.IsNullOrEmpty(this.GitHubToken))
                 {
-                    Logger.Trace("No cached token found.");
-                    Logger.DebugLocalized(nameof(Resources.GitHubAccountMustBeLinked_Message));
-                    Logger.DebugLocalized(nameof(Resources.ExecutionPaused_Message));
-                    Console.WriteLine();
-                    token = await GitHubOAuthLoginFlow();
-                    if (string.IsNullOrEmpty(token))
-                    {
-                        // User must've cancelled OAuth flow, we can't proceed successfully
-                        Logger.WarnLocalized(nameof(Resources.NoTokenResponse_Message));
-                        return false;
-                    }
+                    Logger.WarnLocalized(nameof(Resources.RepeatedUseWarning_Message));
+                    Logger.WarnLocalized(nameof(Resources.AvoidRateLimitWarning_Message));
+                }
+            }
+            else
+            {
+                await this.StoreTokenInCache();
```

Comment:
This is a behavior change, before we only cached the token from oauth, not from the -t parameter. I can't remember why we decided to do that exactly, but we should make sure we're intentionally changing that

Reply:
I am aware of this behavior change. I probably should have run it by you first to be sure. Are you okay with this change? If the -t <token> they use is valid, I think it should be safe to cache it for the user for future use. Let me know if you are/aren't okay with that

Reply:
I'm fine with it, adding @kevinla and @denelon to make sure they are too

Reply:
I don't have any issues keeping that token for future use. We could address it in the future if anyone has a complaint. I don't think there are too many cases where people would be juggling different GitHub identities.

Reply:
@KevinLaMS  and I talked offline. We decided it would be better to not cache the token automatically when provided as a -t token as this could be a behavior that the user doesn't want/isn't expecting. The token command should be sufficient for storing tokens and this PR fixes the bug where other commands weren't consuming the cached token when calling github apis. I have reverted this change of behavior.